### ISSUE TITLE
cannot find module '@cljs-oss/module-deps'

### ISSUE THREAD

**Issue by mtruyens** [bhauman/figwheel-main#21]
Hi, 

I love the new Figwheel, but I'm occasionally receiving the following error warning: 

```
[Figwheel] Compiling build dev to "resources/public/js/app/dev/app.js"
internal/modules/cjs/loader.js:583
    throw err;
    ^

Error: Cannot find module '@cljs-oss/module-deps'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:581:15)
    at Function.Module._load (internal/modules/cjs/loader.js:507:25)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at [eval]:8:13
    at Script.runInThisContext (vm.js:91:20)
    at Object.runInThisContext (vm.js:298:38)
    at Object.<anonymous> ([eval]-wrapper:6:22)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at evalScript (internal/bootstrap/node.js:527:27)
[Figwheel] Failed to compile build dev in 2.779 seconds.
[Figwheel] Compile Exception: 
```

This (somewhat randomly) seems to occurs after several successful change/reload cycles. Doing a (figwheel.main/reset) does not help, it seems necessary to kill the process and restart it. 

I'm using Leiningen, on Mac, Clojurescript 1.10.339.

Any idea?

**Comment by mtruyens** [comment]
Addendum: 

just a hunch, but I have seen this error two times several minutes after a reader-error occurs — e.g., when I erroneously inserted a single colon somewhere:

(dispatch [ : ])

Even so, the "cannot find module" error does not immediately occur, only after a while. 

Perhaps also good to note, is that my /target folder got corrupted twice today (having installed Figwheel-main this morning), leading to very strange errors in the browser — something I only very infrequently encountered in the old Figwheel. Only when removing the entire folder, then recompiling, and performing a hard-clean of all browser caches, did the problem go away.

**Comment by bhauman** [comment]
This error seems to intermittently crop up.  

My belief is that this originates in the CLJS Compiler and as such if you create a minimal case you should be able to reproduce it with `cljs.main`.   I think the compiler is corrupting its own state somehow when this happens.

I would like to know what npm related compiler options you are using.

**Comment by mtruyens** [comment]
Thanks! 

In the meantime, I had to switch back to the old Figwheel, because some macros (embedded in CLJC files) that I use for Spec-checking were somehow not being compiled, even though I received no warnings at all. 

Anyway, here is the contents of dev.cljs I used for Figwheel-main. (I am aware of the issues associated with recompile-dependents being set to false, but this hardly ever resulted in issues under the old Figwheel. Maybe also good to note is that I have a hybrid CLJ/CLJS project, with the ClojureScript being loaded through a separate ring-server)

```
{:main                 sx.cljs.app
 :output-to            "resources/public/js/app/dev/app.js"
 :output-dir           "target-cljs/cljsbuild/js/out/dev"
 :parallel-build       true 
 :pretty-print         true
 :asset-path           "/js/out/dev" 
 :optimizations        :none
 :verbose              false
 :source-map           true
 :source-map-timestamp true
 :infer-externs        true
 :npm-deps             false
 :recompile-dependents false
 :foreign-libs         [{:file           "resources/public/js/bundle/dev/bundle.js"
                         :provides       ["react" "react-dom" "create-react-class" "react-dom/server"]
                         :global-exports {react              React
                                          react-dom          ReactDOM
                                          create-react-class createReactClass
                                          react-dom/server   ReactDOMServer}}]
 }

```

**Comment by bhauman** [comment]
The macros in the CLJC file is a separate issue. Let's address that separately. 

Yes this problem as cropped up with `:npm-deps false` before.

**Comment by bhauman** [comment]
OK now I'm tracking this error more closely.

The error still gets printed but does not prevent reloads from occurring in "0.1.8-SNAPSHOT".

**Comment by bhauman** [comment]
this is fixed with 
https://github.com/bhauman/figwheel-core/commit/77bf8beb43d94038c246e41637b7df0d7c6fad26

Sorry I didn't get around to this sooner. This fix will be available in 0.1.8 when its released.

**Comment by mtruyens** [comment]
Many thanks (for this bug & Figwheel in general!). Will try it immediately when 0.1.8 is indeed released.

**Comment by bhauman** [comment]
I released 0.1.8-SNAPSHOT moments ago if your interested.

**Comment by mtruyens** [comment]
After two days of intensive use, I can confirm that the bug has vanished. Thanks!

**Comment by glittershark** [comment]
This is happening again for me with 0.2.11
